bin/github-review.sh

```diff
@@ -45,53 +41,130 @@ if ! [[ $1 =~ ^[0-9]+$ ]]; then
     exit 1
 fi
 pr=$1
-if ! [[ $2 =~ ^[0-9]+$ ]]; then
-    echo "ERROR: $2 is not a valid Chain ID number" 1>&2
+prChainID="$(gh pr view $pr | sed -nE 's/.*Chain_ID: ([0-9]+).*/\1/p')"
+if ! [[ $prChainID =~ ^[0-9]+$ ]]; then
+    echo "ERROR: $prChainID is not a valid Chain ID number" 1>&2
     usage
     exit 1
 fi
-rpc=$3
+chainlistURL="https://chainlist.org/chain/$prChainID"
+if [[ "$(curl -s "$chainlistURL")" == 'nope' ]]; then
+    echo "ERROR: Chainlist URL $chainlistURL doesn't exist" 1>&2
+    usage
+    exit 1
+fi
+rpc="$(gh pr view $pr | sed -nE 's/.*RPC_URL: (https?:\/\/[^ ]+).*/\1/p')"
 chainid="$(cast chain-id --rpc-url $rpc)"
-if [[ $chainid != $2 ]]; then
-    echo "ERROR: RPC $rpc doesn't match chain ID $2" 1>&2
+if [[ $chainid != $prChainID ]]; then
+    echo "ERROR: RPC $rpc doesn't match chain ID $prChainID" 1>&2
     usage
     exit 1
 fi
-version=$4
+version="$(gh pr view $pr | sed -nE 's/.*Contract_Version: (1\.[3-4]\.[0-1]).*/\1/p')"
 versionFiles=(src/assets/v$version/*.json)
 if [[ ${#versionFiles[@]} -eq 0 ]]; then
     echo "ERROR: Version $version doesn't exist" 1>&2
     usage
     exit 1
 fi
 
+# Fetching PR and checking if other files are changed or not.
+echo "Checking changes to other files"
+if [[ -n "$(gh pr diff $pr --name-only | grep -v -e 'src/assets/v'$version'/.*\.json')" ]]; then
+    echo "ERROR: PR contains changes in files other than src/assets/v$version/*.json" 1>&2
+    echo "Changed files:"
+    echo "$(gh pr diff $pr --name-only | grep -v -e 'src/assets/v'$version'/.*\.json')" 1>&2
+    exit 1
+fi
+
+echo "Checking changes to assets files"
+fileLineChangeJSON=$(gh pr view $pr --json files) # This line fetches the JSON output of the files changed in the PR
+edgeCase=0
+echo "$fileLineChangeJSON" | jq -r '.files[] | "\(.path) \(.additions) \(.deletions)"' | while read -r line; do
+    path=$(echo $line | cut -d ' ' -f1)
+    additions=$(echo $line | cut -d ' ' -f2)
+    deletions=$(echo $line | cut -d ' ' -f3)
+
+    # Now you can perform checks on $additions and $deletions as per your requirements
+    if [[ ($additions == 2 && $deletions == 1) ]]; then
+        edgeCase=1
+    elif [[ ($additions != 1 || $deletions != 0) ]]; then
+        echo "ERROR: $path has invalid changes" 1>&2
+        exit 1
+    fi
+done
+
+if [[ $edgeCase == 1 ]]; then
+    echo "Edge case when adding chain with the highest chain id number"
+    diffPatchSeparated=($(gh pr diff $pr | grep -E '^[+-] '))
+    # Adding three elements at a time together to compare from the output array
+    diffPatch=()
+    for ((i=0; i<${#diffPatchSeparated[@]}; i+=3)); do
+        diffPatch+=("${diffPatchSeparated[i]} ${diffPatchSeparated[i+1]} ${diffPatchSeparated[i+2]}")
```

Comment:
Doesn't this depend on how many before lines and after lines are included in the diff?

Reply:
Yes, but we already know that if it enters this `if` branch, there are only 2 additions and 1 subtraction. With `grep` we only take the actual lines added and deleted in the diff patch. And if it doesn't follow the intended pattern (based on the check we have written), then it errors out, as it should be.